packages/core/components/FileThumbnail/index.tsx

```diff
@@ -11,12 +12,29 @@ interface Props {
     uri?: string;
     height?: number | string;
     width?: number | string;
+    loading?: boolean;
 }
 
 /**
- * Displays the thumbnail for a file.
+ * Displays the thumbnail for a file, or a spinner if loading, or a placeholder if missing.
  */
 export default function FileThumbnail(props: Props) {
+    // If loading, render the spinner instead of the image
+    if (props.loading) {
+        return (
+            <div
+                className={classNames(
+                    props.className,
+                    styles.fileThumbnail,
+                    styles.thumbnailPlaceholder
+                )}
+                style={{ height: props.height, width: props.width }}
```

Comment:
Should this default to being a square if one of the fields is present but not the other? Like so:
```
style={{ height: props.height || props.width, width: props.width || props.height }}
```

Reply:
resolved in [478fbd2](https://github.com/AllenInstitute/biofile-finder/pull/268/commits/478fbd2d80b96884e30648830079e4285557fbf1)